### PR TITLE
remove type_ attribute and use type instead

### LINKED ISSUE #16: Change action match to be based on type rather than instance variable
The `Action` type currently has a `type_` attribute which is used in `match`, this is duplicate and can be replaced using a match on `type(action)`

### PR DESCRIPTION
**Pull request by jdkandersson**
Changes the match statements to be based on the `type` function rather than the `type_` attribute, closes #16

### CODE REVIEW

**Review by yanksyoon — DISMISSED** [review]
LGTM!

**Review by arturo-seijas — APPROVED** [review]
LGTM

File: src/action.py [hunk 1]
```diff
@@ -276,15 +276,19 @@ def _run_index(
...
-            url=DRY_RUN_NAVLINK_LINK if action.type_ == types_.ActionType.CREATE else action.url,
+            url=(
+                DRY_RUN_NAVLINK_LINK
+                if isinstance(action, types_.CreateIndexAction)
+                else action.url
+            ),
             result=types_.ActionResult.SKIP,
             reason=DRY_RUN_REASON,
         )
         logging.info("report: %s", report)
```
Comment (yanksyoon) [inline src/action.py:287]: Unrelated to PR, but may I know when to use "%s" and f-strings?

**Review by yanksyoon — APPROVED** [review]
LGTM

File: src/action.py (hunk 1 above)
Comment (arturo-seijas) [reply src/action.py:287]: https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#f-strings

**Review by merkata — APPROVED** [review]

**Pull request merged**